examples/nirf/specan-nr-txp-example.py

```diff
@@ -0,0 +1,343 @@
+        nirfmxnr_types.TXPFetchPowerTraceRequest(
+            instrument=instr,
+            selector_string="",
+            timeout=10.0,
+        )
+    )
+    check_for_warning(txp_fetch_power_trace_response, instr)
+    x0 = txp_fetch_power_trace_response.x0
+    dx = txp_fetch_power_trace_response.dx
+    power = txp_fetch_power_trace_response.power
+
+    print("\n------------NR TXP Measurement------------\n")
+    print(f"Average Power Mean (dBm) : {avearge_power_mean}")
+    print(f"Peak Power Maximum (dBm) : {peak_power_maximum}")
+
+    # Configure and test SpecAn TXP
+    specanclient.SetAttributeString(
+        nirfmxspecan_types.SetAttributeStringRequest(
+            instrument=instr,
+            selector_string="",
+            attribute_id=nirfmxspecan_types.NIRFMXSPECAN_ATTRIBUTE_SELECTED_PORTS,
+            attr_val_raw="",
+        )
+    )
+
+    specanclient.CfgRF(
+        nirfmxspecan_types.CfgRFRequest(
+            instrument=instr,
+            selector_string="",
+            center_frequency=1e9,
+            reference_level=0.00,
+            external_attenuation=0.00,
+        )
+    )
+
+    specanclient.TXPCfgMeasurementInterval(
+        nirfmxspecan_types.TXPCfgMeasurementIntervalRequest(
+            instrument=instr,
+            selector_string="",
+            measurement_interval=1e-3,
+        )
+    )
+
+    specanclient.TXPCfgRBWFilter(
+        nirfmxspecan_types.TXPCfgRBWFilterRequest(
+            instrument=instr,
+            selector_string="",
+            rbw=100e3,
+            rbw_filter_type=nirfmxspecan_types.TXP_RBW_FILTER_TYPE_GAUSSIAN,
+            rrc_alpha=0.1,
+        )
+    )
+
+    specanclient.TXPCfgAveraging(
+        nirfmxspecan_types.TXPCfgAveragingRequest(
+            instrument=instr,
+            selector_string="",
+            averaging_enabled=nirfmxspecan_types.TXP_AVERAGING_ENABLED_FALSE,
+            averaging_count=10,
+            averaging_type=nirfmxspecan_types.TXP_AVERAGING_TYPE_RMS,
+        )
+    )
+
+    ### Fetch SpecAn measurement data ###
+
+    txp_read_response = specanclient.TXPRead(
+        nirfmxspecan_types.TXPReadRequest(
+            instrument=instr,
+            selector_string="",
+            timeout=10.0,
+        )
+    )
+    check_for_warning(txp_read_response, instr)
+
+    minimum_power = txp_read_response.minimum_power
+    maximum_power = txp_read_response.maximum_power
+    peak_to_average_ratio = txp_read_response.peak_to_average_ratio
+    average_mean_power = txp_read_response.average_mean_power
+
+    print("\n------------SpecAn TXP Measurement------------\n")
+    print(f"Average Mean Power(dBm)    {average_mean_power}")
+    print(f"Peak to Average Ratio(dB)  {peak_to_average_ratio}")
+    print(f"Maximum Power(dBm)         {maximum_power}")
+    print(f"Minimum Power(dBm)         {minimum_power}")
+except grpc.RpcError as rpc_error:
+    error_message = rpc_error.details()

```

Comment:
```suggestion
    error_message = str(rpc_error.details() or "")
```
We had to update this line in all the pre-existing examples when the mypy version we were using was bumped to avoid a violation here.